documentation/library_guide/dynamic_selection_api/functions.rst

```diff
@@ -0,0 +1,595 @@
+                             return e; /* MUST return sycl::event */
+                          },
+                          /* any additional args... */ i);  
+      std::cout << "(i == " << i << "): submission done\n"; 
+    }
+  }
+
+The output from this example:
+
+.. code::
+
+  (j == 0): submit to cpu
+  (i == 0): submission done
+  (j == 1): submit to gpu
+  (i == 1): submission done
+  (j == 2): submit to cpu
+  (i == 2): submission done
+  (j == 3): submit to gpu
+  (i == 3): submission done
+
+
+submit and wait using a selection
++++++++++++++++++++++++++++++++++
+
+.. code:: cpp
+
+  namespace oneapi::dpl::experimental {
+    template<typename Selection, typename F, typename... Args> 
+    void submit_and_wait(Selection s, F&& f, Args&&... args);
+  }
+  
+Calls the user function ``f``, passing the unwrapped selection ``s`` and ``args...`` 
+as the arguments. It also implements the necessary calls to report execution 
+information for policies that require reporting.
+
+This function blocks the calling thread until 
+the user function and any work that it submits to the resource
+are complete.
+
+The following example demonstrates the use of function ``submit_and_wait``. 
+The use of ``single_task`` is for syntactic demonstration 
+purposes only; any valid command group or series of command groups can be 
+submitted to the selected queue.
+
+.. code::  cpp
+
+  #include <oneapi/dpl/dynamic_selection>
+  #include <sycl/sycl.hpp>
+  #include <iostream>
+  
+  namespace ex = oneapi::dpl::experimental;
+  
+  int main() {
+    ex::round_robin_policy p{ { sycl::queue{ sycl::cpu_selector_v },  
+                                sycl::queue{ sycl::gpu_selector_v } } };
+  
+    for (int i = 0; i < 4; ++i) {
+      auto s = ex::select(p);
+      ex::submit_and_wait(/* selection object */ s,  
+                          /* user function */
+                          [](sycl::queue q, /* any additional args... */ int j) {
+                             std::cout << "(j == " << j << "): submit to " 
+                                       << ((q.get_device().is_gpu()) ? "gpu\n" : "cpu\n");
+                             auto e = q.single_task([]() { /* do some work */ }); 
+                             return e; /* MUST return sycl::event */
+                          },
+                          /* any additional args... */ i);  
+      std::cout << "(i == " << i << "): submission done\n"; 
+    }
+  }
+
+
+The output from this example:
+
+.. code::
+
+  (j == 0): submit to cpu
+  (i == 0): submission done
+  (j == 1): submit to gpu
+  (i == 1): submission done
+  (j == 2): submit to cpu
+  (i == 2): submission done
+  (j == 3): submit to gpu
+  (i == 3): submission done
+
+Policy Queries
+--------------
+
+Getting the resource options
+++++++++++++++++++++++++++++
+
+.. code:: cpp
+
+  namespace oneapi::dpl::experimental {
+    template<typename Policy, typename... Args> 
+    std::vector<resource_t<Policy>> get_resources(P&& p);
```

Comment:
What's `P&&`? I guess it should be `Policy&&`

Reply:
done. Made this consistent as "Policy" across all uses in all files.